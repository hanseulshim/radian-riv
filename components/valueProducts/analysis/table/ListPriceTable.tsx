import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import { formatPercent } from 'utils'

interface Props {
  listingPrice: {
    '0-90': number
    '91-180': number
    '180-270': number
    '271-365': number
  }
  type: string
}

export default function ListPriceTable({ type, listingPrice }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const tableData = [{ ...listingPrice }]
    setColumns([
      {
        Header: `MEDIAN SOLD PRICE AS % OF ${type} LIST PRICE`,
        columns: [
          {
            Header: 'Most Recent 90 Days',
            accessor: row => formatPercent(row['0-90']),
            align: 'right',
            width: 130
          },
          {
            Header: 'Prior 91 to 180 Days',
            accessor: row => formatPercent(row['91-180']),
            align: 'right',
            width: 100
          },
          {
            Header: 'Prior 181 to 270 Days',
            accessor: row => formatPercent(row['180-270']),
            align: 'right',
            width: 110
          },
          {
            Header: 'Prior 271 to 360 Days',
            accessor: row => formatPercent(row['271-365']),
            align: 'right'
          }
        ]
      }
    ])
    setData(tableData)
  }, [listingPrice])

  return (
    <div style={{ marginRight: type === 'FINAL' ? '1em' : 0 }}>
      <Table columns={columns} data={data} fetchData={fetchData} />
    </div>
  )
}
