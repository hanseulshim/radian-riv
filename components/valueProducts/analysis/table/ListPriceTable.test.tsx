import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ListPriceTable from './ListPriceTable'

describe('ListPriceTable', () => {
  let component
  beforeAll(() => {
    const props = {
      listPrice: {
        '0-90': 1,
        '91-180': 1,
        '180-270': 1,
        '271-365': 1
      },
      type: 'test'
    }
    component = shallow(<ListPriceTable {...props} />)
  })
  it('get table container', () => {
    expect(component.find('div')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
