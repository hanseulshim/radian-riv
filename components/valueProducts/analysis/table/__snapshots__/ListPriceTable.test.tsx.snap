// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPriceTable should match snapshot 1`] = `
<div
  style={
    Object {
      "marginRight": 0,
    }
  }
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
  />
</div>
`;
